trace-reader/src/main.rs

```diff
@@ -0,0 +1,105 @@
+use anyhow::Result;
+use clap::Parser;
+use common::{DigitizerId, FrameNumber};
+use rand::{seq::IteratorRandom, thread_rng};
+use rdkafka::producer::FutureProducer;
+use std::path::PathBuf;
+
+mod loader;
+mod processing;
+use loader::load_trace_file;
+use processing::dispatch_trace_file;
+
+#[derive(Debug, Parser)]
+#[clap(author, version, about)]
+struct Cli {
+    ///Kafka message broker, should have format `host:port`, e.g. `localhost:19092`
+    #[clap(long)]
+    broker: String,
+
+    ///Optional Kafka username
+    #[clap(long)]
+    username: Option<String>,
+
+    ///Optional Kafka password
+    #[clap(long)]
+    password: Option<String>,
+
+    ///Name of the Kafka consumer group
+    #[clap(long)]
+    consumer_group: String,
+
+    ///The Kafka topic that trace messages are produced to
+    #[clap(long)]
+    trace_topic: String,
+
+    ///Relative path to the .trace file to be read
+    #[clap(long)]
+    file_name: PathBuf,
+
+    ///The frame number to assign the message
+    #[clap(long, default_value = "0")]
+    frame_number: FrameNumber,
+
+    ///The digitizer id to assign the message
+    #[clap(long, default_value = "0")]
+    digitizer_id: DigitizerId,
+
+    ///The number of trace events to read. If zero, then all trace events are read
+    #[clap(long, default_value = "1")]
+    number_of_trace_events: usize,
+
+    ///If set, then trace events are sampled randomly with replacement, if not set then trace events are read in order
+    #[clap(long, default_value = "false")]
+    random_sample: bool,
+}
+
+#[tokio::main]
+async fn main() -> Result<()> {
+    env_logger::init();
+
+    let args = Cli::parse();
+
+    let client_config =
+        common::generate_kafka_client_config(&args.broker, &args.username, &args.password);
+
+    let producer: FutureProducer = client_config
+        .create()
+        .expect("Failed to create Kafka Producer");
+
+    let trace_file = load_trace_file(args.file_name).expect("Failed to Load Trace File");
```

Comment:
The message given to `expect` should be a summary of what should happen that did not, i.e. you expected A, but A is not the case.

e.g.
```suggestion
    let trace_file = load_trace_file(args.file_name).expect("trace file should be loaded");
```